exonum/src/runtime/execution_context.rs

```diff
@@ -46,6 +52,8 @@ pub struct ExecutionContext<'a> {
     dispatcher: &'a Dispatcher,
     /// Depth of the call stack.
     call_stack_depth: u64,
+    /// Flag indicating an error occurred during the child call.
+    has_child_call_error: &'a mut bool,
```

Comment:
To be honest, a mutable reference to `bool` is a really weird thing. I'd suggest to encapsulate it into a separate type with meaningful method names.

Reply:
As we've discussed with @alekseysidorov, it could be possible to encapsulate this flag (i.e., use `bool` instead of `&mut bool`) if a method is added to `ExecutionContext` to transform an execution result:

```rust
impl ExecutionContext<'_> {
    pub(crate) fn coerce_result(&self, res: Result<(), ExecutionError>) -> Result<(), ExecutionError> {
        if self.has_child_call_error {
            Err(CoreError::InvalidCall.into())
        } else {
            res
        }
    }
}
```

Would it be an improvement?